fw/http.c

```diff
@@ -4406,7 +4406,7 @@ tfw_h2_resp_add_loc_hdrs(TfwHttpResp *resp, const TfwHdrMods *h_mods,
 		const TfwHdrModsDesc *desc = &h_mods->hdrs[i];
 		int r;
 
-		if (test_bit(i, mit->found) || !TFW_STR_CHUNK(desc->hdr, 1))
+		if (test_bit(i, mit->found) || (TFW_STR_CHUNK(desc->hdr, 1) == NULL))
```

Comment:
I checked only this place and didn't get why the compiler says that the expression is always true: `h_mods->hdrs[i]` might be plain or now and it might has chunks or not.

Also `!(...)` looks valid C expression for me and I don't understand why `(...) == NULL` changes anything? I didn't have time to play with the compiler, but the fix looks confusing: we have a lot of similar places and these particular 2 places can be changed back. Maybe, if the fix is correct, it'd be better to introduce a specific macro?

Reply:
Yeah, it's strange since there are many places quite like this in code and the compiler emits warning only here.

Reply:
Looks like compiler interprets the expression `(s)->chunks + (c)` from macro `__TFW_STR_CH` as `!(NULL + (c))` or `!(Pointer + (c))`, these expression will have same result and condition doesn't have a sense. However, we have a `(c) >= (s)->nchunks` in `TFW_STR_CHUNK` that prevent case with `NULL + (c)`, but compiler "ignores" this.

I think, we can rewrite:
`#define __TFW_STR_CH(s, c)	((s)->chunks + (c))`
to
`#define __TFW_STR_CH(s, c)	(&(s)->chunks[(c)])`
With `-O2` it will be compiled to same code. But I don't test it.

Reply:
Unfortunately, the proposed fix results in the exact same error.